odxtools/minmaxlengthtype.py

```diff
@@ -167,9 +168,13 @@ def convert_bytes_to_internal(self,
             value, byte_pos = self._extract_internal_value(
                 decode_state.coded_message,
                 byte_position=cursor_pos,
-                bit_position=bit_position,
+                bit_position=0,
                 bit_length=8 * byte_length,
                 base_data_type=self.base_data_type,
                 is_highlow_byte_order=self.is_highlow_byte_order,
             )
-            return value, byte_pos
+
+            decode_state.cursor_position = byte_pos
+            decode_state.cursor_bit_position = None
```

Comment:
Maybe add a `goto(byte_pos, bit_pos = 0)` function to `decode_state` ?

Reply:
I've tried this as a prototype, but I was not really happy with it because it obfuscated what was happening too much. That said, how do you interpret the following sentence from section 7.3.5.2 of the ASAM version of the spec:  

```
The parameter starts then at the byte edge next to the end of the last parameter.
```

does "last parameter" mean
a) the parameter which is directly before the one currently decoded in the list of parameters
b) the parameter which encompasses the byte of the PDU furthest to the back which has been seen so far
c) the parameter specified last in the list

IMO, (c) does not make sense, we currently use interpretation (b), but the more I think about it, (a) makes more sense (and, as a bonus, is easier to implement)...

Reply:
Both A and B are valid interpretations given how vague the spec is, and would give identical results given sane enough PDX file that does order the parameters by desired position.

That being said, the specs didn't use the term "previous parameter" which is more common, making option B more likely.

Reply:
okay. would you support switching the interpretation to option (a) in the future anyway? (it simplifies things considerably...)

Reply:
I have not seen a PDX so far, where choosing A or B matters, if I ran into issues I will let you know.